packages/web-components/cypress/e2e/smart-camera-web-agent-mode.cy.js

```diff
@@ -31,68 +31,23 @@ variants.forEach(({ name, suffix }) => {
 
       cy.get('smart-camera-web')
         .shadow()
-        .find('selfie-capture')
+        .find('selfie-capture-wrapper')
         .should('be.visible');
 
-      cy.get('smart-camera-web')
-        .shadow()
-        .find('selfie-capture')
-        .shadow()
-        .should('contain.text', 'Agent Mode On');
+      // Wait for component to load and setup
+      cy.wait(3000);
 
       cy.get('smart-camera-web')
         .shadow()
-        .find('selfie-capture')
-        .shadow()
-        .find('#switch-camera')
-        .click();
-
-      cy.get('smart-camera-web')
-        .shadow()
-        .find('selfie-capture')
-        .shadow()
-        .should('contain.text', 'Agent Mode Off');
-
-      cy.get('smart-camera-web')
+        .find('selfie-capture-wrapper')
         .shadow()
         .find('selfie-capture')
         .shadow()
         .find('#switch-camera')
```

Comment:
Looks like we are only checking the existence of the agent mode button but not if it switches

Reply:
The switching only works if we really have multiple cameras, with at least one being a back camera and one being a front camera. When clicked, we also have to request the camera from the browser and verify that what we got is what we requested before updating the button. I've yet to find a way to get this behavior to work in cypress tests.